keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -402,129 +404,124 @@ type AdminNetworkPolicy struct {
   Subject     AdminNetworkPolicySubject
 
   // List of Ingress rules to be applied to the selected objects.
-  // A total of 100 rules will be allowed per each network policy instance, 
-  // this rule count will be calculated as the total summation of the 
-  // Ingress and Egress rules in a single AdminNetworkPolicy Instance. 
+  // A total of 100 rules will be allowed per each network policy instance,
+  // this rule count will be calculated as the total summation of the
+  // Ingress and Egress rules in a single AdminNetworkPolicy Instance.
   Ingress     []AdminNetworkPolicyIngressRule
 
   // List of Egress rules to be applied to the selected objects.
-  // A total of 100 rules will be allowed per each network policy instance, 
-  // this rule count will be calculated as the total summation of the 
-  // Ingress and Egress rules in a single AdminNetworkPolicy Instance. 
+  // A total of 100 rules will be allowed per each network policy instance,
+  // this rule count will be calculated as the total summation of the
+  // Ingress and Egress rules in a single AdminNetworkPolicy Instance.
   Egress      []AdminNetworkPolicyEgressRule
 }
 
-// AdminNetworkPolicySubject defines what objects the policy selects. 
-// Exactly one of the selector pointers should be set.  
+// AdminNetworkPolicySubject defines what objects the policy selects.
+// Exactly one of the `Namespaces` or `NamespacedPods` pointers should be set.
 type AdminNetworkPolicySubject struct {
-  NamespaceSelector         *metav1.LabelSelector
-  NamespaceAndPodSelector   *NamespaceAndPodSelector
+  Namespaces        *metav1.LabelSelector
+  NamespacedPods    *NamespacedPodSubject
 }
 
-// NamespaceAndPodSelector allows the user to select a Namespace(s) and optionally 
-// select a given set of pod(s) in that namespace(s)
-type NamespaceAndPodSelector struct {
-  // This field follows standard label selector semantics; if present but empty, 
-  // it selects all Namespaces.  
-  NamespaceSelector   metav1.LabelSelector
-
-  // Used to explicitly select pods within a namespace; if present but empty, 
-  // it selects all Pods.  
-  PodSelector         metav1.LabelSelector
+// NamespacedPodSubject allows the user to select a given set of pod(s) in
+// selected namespace(s)
+type NamespacedPodSubject struct {
+  NamespaceSelector  *metav1.LabelSelector
+  PodSelector        *metav1.LabelSelector
 }
 
-// AdminNetworkPolicyIngressRule describes an action to take on a particular 
-// set of traffic destined for pods selected by an AdminNetworkPolicy's 
-// Subject field. The traffic must match both l4Selector and from. 
+// AdminNetworkPolicyIngressRule describes an action to take on a particular
+// set of traffic destined for pods selected by an AdminNetworkPolicy's
+// Subject field. The traffic must match both l4Selector and from.
 type AdminNetworkPolicyIngressRule struct {
   // Name is an identifier for this rule.
   // +optional
```

Comment:
We still need a MAX-length in the name (not really sure how to do this)

Reply:
Done. And I double-checked that this can be done via openAPI